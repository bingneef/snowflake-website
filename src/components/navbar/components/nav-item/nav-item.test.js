import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import NavItem from "./nav-item";

describe("NavItem", () => {
  it("renders without error", () => {
    const div = document.createElement("div");
    ReactDOM.render(<NavItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("matches the snapshot", () => {
    const component = renderer.create(<NavItem />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches the snapshot on inverse", () => {
    const component = renderer.create(<NavItem inverse={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches the snapshot on active", () => {
    const component = renderer.create(<NavItem active={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
