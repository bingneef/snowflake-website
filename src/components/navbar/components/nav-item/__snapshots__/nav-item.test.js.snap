// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem matches the snapshot 1`] = `
<a
  className="navItem"
/>
`;

exports[`NavItem matches the snapshot on active 1`] = `
<a
  className="navItem isActive"
/>
`;

exports[`NavItem matches the snapshot on inverse 1`] = `
<a
  className="navItem inverse"
/>
`;
